source/developing-volttron/development-environment/vscode/index.rst

```diff
@@ -0,0 +1,83 @@
+.. _VSCode-Dev-Environment:
+
+===============================
+VSCode Development Environment
+===============================
+
+Before setting up the VSCode development environment, ensure the following prerequisites are met:
+
+  1. A Linux environment on WSL2 with necessary software installed. Refer to :ref:`Install WSL2 Ubuntu environment <Install-WSL2-Ubuntu>` for setting up a compatible Linux environment.
+  2. Your own fork of the relevant VOLTTRON repositories cloned in your WSL2 environment. 
+
+VSCode is a versatile Integrated Development Environment (IDE) that supports Python development and offers extensive debugging tools and GitHub integration. It is widely used for developing VOLTTRON projects.
+
+Microsoft provides VSCode as a free download from https://code.visualstudio.com/
+
+Install VSCode
+==============
+VSCode can be installed directly on your Windows host and configured to connect to code within the WSL2 environment.
+
+**Installing VSCode on Windows:**
+- Download the VSCode installer for Windows from the official site and follow the installation prompts.
+- Once installed, open VSCode and install the **Remote - WSL** extension from the marketplace. This extension allows VSCode to connect and interact with Linux distributions installed via WSL2.
+
+**Configuring WSL2 with VSCode:**
+- Open VSCode.
+- Use the command palette (`Ctrl+Shift+P`) and select **Remote-WSL: New Window**. This opens a new VSCode window directly connected to your WSL2 environment.
```

Comment:
how do you know which wsl2 container to connect to?

Reply:
Will add more notes: For more detailed guidance and latest updates, refer to the official Visual Studio Code documentation on [Working with multiple WSL2 distributions](https://code.visualstudio.com/docs/remote/wsl#_working-with-multiple-linux-distributions).